### PR TITLE
fix(api): derive suite verification from active entitlements

### PR DESCRIPTION
**Pull request by Yeusepe**
## Summary
- Derive Suite OAuth-JWT verification status from the subject's active entitlement results.
- Add a regression test proving a zero-entitlement subject returns `verified: false`.
- Register the invariant and focused route test in the production regression gate.

## Verification
- `bun test apps/api/src/routes/suite.test.ts` (RED before fix, GREEN after)
- `bun run lint`
- `bun run typecheck`
- `bun run test:external-integrations`
- `bun run test:ops`, `bun run test:fast:ci`, `bun run test:boundary-mocks`, `bun run test:convex`, and `bun run test:api:integration`

`bun audit` remains blocked by the pre-existing moderate `@better-auth/oauth-provider` advisory. No dependency or lockfile changes were made. The top-level `bun run test:ci` wrapper completed all stages but did not exit locally after the final passing API integration stage; the constituent stages above pass independently.

### CODE REVIEW

**Comment by Yeusepe** [comment]
@codex review

Do an ADVERSARIAL review of this PR, not a rubber stamp. Assume there is a real bug and try to find it.
Review the changed files only, but open the callees they touch. For each point, cite `file:line` and a
concrete failing input -> wrong outcome -> the fix. If a section is genuinely clean, say what you checked
and why it is safe. Do not approve unless you actually traced these:

1. Threat-model each changed request path. Simulate hostile input: missing / empty / wrong-type fields,
   oversized payloads, another tenant's ids or authUserId, forged signatures. Flag anything that 500s
   instead of 4xx, leaks another tenant's data, or fails open (authorizes when it should deny).
2. Data flow across the api<->Convex boundary. For every changed Convex call, confirm the args EXACTLY
   match the callee's validator (names, shape, required actor / apiSecret / authUserId) and the response
   is read with the correct shape (object vs bare id, the `{ found, subject }` wrapper, cursor/next-cursor
   keys). Name the callee file:line.
3. Test adequacy. For each behavior this PR fixes or adds, point to the test that would FAIL if that
   behavior regressed. If there is no such test, call it out as unverified - do not assume coverage.
4. Secrets & logging. Does any changed error/log path emit a token, key, signing secret, or a resource id
   that must not be logged? Prefer a generic message + correlation id.
5. Redaction / sanitization changes specifically: confirm they do not now under-redact (leak) or
   over-redact (drop data) for arrays, nested objects, and primitives.

Rank findings by severity. Distinguish real defects from style nits. Be specific enough that each finding
is independently verifiable.

File: apps/api/src/routes/suite.ts [hunk 1]
```diff
@@ -126,7 +126,7 @@ export async function getVerificationStatus(
   );
 
   return jsonResponse({
-    verified: true,
+    verified: products.length > 0,
```
[the remark this answers is not in the document]
Comment (Yeusepe) [reply apps/api/src/routes/suite.ts:138]: Fixed in 278100e6. Suite status now carries the resolved subject eligibility and requires `subject.status === 'active'` as well as an active entitlement before returning `verified: true`. Added a regression for a suspended subject with an active entitlement, which now returns `verified: false`.

File: apps/api/src/routes/suite.ts [hunk 2]
```diff
@@ -126,7 +127,7 @@ export async function getVerificationStatus(
   );
 
   return jsonResponse({
-    verified: true,
+    verified: subjectResult.isActive && products.length > 0,
```
[the remark this answers is not in the document]
Comment (Yeusepe) [reply apps/api/src/routes/suite.ts]: Fixed in 0ad7c564. Inactive Suite subjects now short-circuit before entitlement loading and return `{ verified: false, products: [] }`. The suspended-subject regression now expects an empty product list and asserts only the subject lookup runs. `bun test apps/api/src/routes/suite.test.ts` passes (2/2).
